.github/workflows/yocto-build-deploy.yml

```diff
@@ -0,0 +1,1174 @@
+
+          # Set Leviathan root location
+          echo "LEVIATHAN_ROOT=${WORKSPACE}/layers/meta-balena/tests/leviathan" >> $GITHUB_ENV
+
+          # Set suites location
+          echo "LEVIATHAN_TEST_SUITE=${WORKSPACE}/layers/meta-balena/tests/suites" >> $GITHUB_ENV
+
+      # Leviathan expects config.js in a certain place. It also expects a specific folder structure
+      # These are done in a separate step as you can't define an env var in the $GITHUB_ENV file and use it in the same step
+      - name: Setup Leviathan Directories
+        run: |
+          # Create workspace directory
+          mkdir -p ${LEVIATHAN_WORKSPACE}
+
+          # Copy config.js to leviathan workspace
+          cp ${LEVIATHAN_TEST_SUITE}/${{ matrix.TEST_SUITE }}/config.js ${LEVIATHAN_WORKSPACE}/
+
+          # Create reports folder
+          echo "LEVIATHAN_REPORTS=${LEVIATHAN_WORKSPACE}/reports" >> $GITHUB_ENV
+
+      # Images need to end up in workspace folder and need to have correct names as well
+      - name: Fetch artifacts from build job
+        uses: actions/download-artifact@v4
+        with:
+          name: build-artifacts
+          path: ${{ env.LEVIATHAN_WORKSPACE }}
+
+      # Image was uploaded uncompressed and Leviathan expects the image in a certain place and with a certain name
+      - name: Prepare image
+        run: |
+          mkdir -p ${LEVIATHAN_REPORTS}
+
+          # The balena.img file is downloaded to ${LEVIATHAN_WORKSPACE}/image/balena.img
+          # Moving it to where the meta-balena config.js expects
+          mv ${LEVIATHAN_WORKSPACE}/image/balena.img ${LEVIATHAN_WORKSPACE}
+
+          sudo apt update
+          sudo apt install -y gzip
+          gzip ${LEVIATHAN_WORKSPACE}/balena.img
+
+      # https://github.com/balena-os/leviathan/blob/master/action.yml
+      - name: BalenaOS Leviathan Tests
+        uses: balena-os/leviathan@282fd606611a023795447d9ad71b5155ea5c0f83
+        env:
+          BALENACLOUD_API_KEY: ${{ env.BALENACLOUD_API_KEY }}
+          BALENACLOUD_API_URL: ${{ vars.BALENA_URL }}
+          BALENACLOUD_APP_NAME: ${{ inputs.BALENACLOUD_APP_NAME }}
+          BALENACLOUD_ORG: ${{ inputs.BALENACLOUD_ORG }}
+          BALENACLOUD_SSH_PORT: ${{ vars.BALENACLOUD_SSH_PORT }}
+          BALENACLOUD_SSH_URL: ${{ vars.BALENACLOUD_SSH_URL }}
+          DEVICE_TYPE: ${{ matrix.DEVICE_TYPE }}
+          LEVIATHAN_ROOT: ${{ env.LEVIATHAN_ROOT }}
+          QEMU_CPUS: 4
+          QEMU_MEMORY: "1G"
+          REPORTS: ${{ env.LEVIATHAN_REPORTS }}
+          SUITES: ${{ env.LEVIATHAN_TEST_SUITE }}
+          WORKER_TYPE: ${{ matrix.WORKER_TYPE }}
+          WORKSPACE: ${{ env.LEVIATHAN_WORKSPACE }}
+          TEST_SUITE: ${{ matrix.TEST_SUITE }}
+
+      # Only to run when images are signed, have to run secureboot tests
+      # Make sure it has all the same env variables as the job above
+      - name: "[Secureboot] BalenaOS Leviathan Tests"
+        if: inputs.sign-image == true
```

Comment:
I am confused here. Doesn't the step before this one need `if: inputs.sign-image == false`?

At least to me it seems that either:

* We are wasting resources running the tests twice for secure boot images; or
* The two additional variables we define here trigger some more complex magic on Leviathan, in which case some additional comments could avoid future misunderstandings.

Reply:
good catch, thats correct

Reply:
True that and additional comments are needed which I will add. It is definitely some magic that Leviathan does, good catch